src/components/dropdown/dropdown.js

```diff
@@ -620,14 +620,15 @@ class Dropdown extends React.Component {
 
     const results = options.map((option) => {
       let klass = className;
+      const optionId = String(option.id);
 
       // add highlighted class
-      if (highlighted == option.id) {
+      if (String(highlighted) === optionId) {
```

Comment:
I think we left this as it may cause a breaking change. These may have to be ignored for now and the change made in CarbonV2